modules/openapi-generator/src/main/resources/typescript-axios/modelEnum.mustache

```diff
@@ -5,14 +5,20 @@ import {EnumMetadata} from '../metadata'
     * {{{description}}}
     */
 {{/description}}
+{{#vendorExtensions.x-enum-as-union}}
+export type {{classname}} = {{#allowableValues}}{{#enumVars}}{{{value}}}{{^-last}} | {{/-last}}{{/enumVars}}{{/allowableValues}};
+{{/vendorExtensions.x-enum-as-union}}
+{{^vendorExtensions.x-enum-as-union}}
 export enum {{classname}} {
 {{#allowableValues}}
 {{#enumVars}}
     {{{name}}} = {{{value}}}{{^-last}},{{/-last}}
 {{/enumVars}}
 {{/allowableValues}}
 }
+{{/vendorExtensions.x-enum-as-union}}
 
+{{! TODO: create a LiteralUnionMetadata for x-enum-as-union? }}
```

Comment:
@stephen-leicht I think I need to learn more about how this metadata is used to generate it properly. Rn it's just a placeholder `values: {}` (same as before this PR, as `Literal` types have no `x-enum-metadata`)

Happy to tackle this in a follow up PR